src/evaluate.cpp

```diff
@@ -1085,12 +1085,13 @@ Value Eval::evaluate(const Position& pos) {
   Value v;
   // Deciding between classical and NNUE eval (~10 Elo): for high PSQ imbalance we use classical,
   // but we switch to NNUE during long shuffling or with high material on the board.
-  bool useClassical = (pos.this_thread()->depth > 9 || pos.count<ALL_PIECES>() > 7) &&
-          abs(eg_value(pos.psq_score())) * 5 > (856 + pos.non_pawn_material() / 64) * (10 + pos.rule50_count());
+  bool useClassical = !useNNUE ? true 
+  : (pos.this_thread()->depth > 9 || pos.count<ALL_PIECES>() > 7) 
+    && abs(eg_value(pos.psq_score())) * 5 > (856 + pos.non_pawn_material() / 64) * (10 + pos.rule50_count());
```

Comment:
This could just be written as
```C++
bool useClassical = !useNNUE ||
                      (   (pos.this_thread()->depth > 9 || pos.count<ALL_PIECES>() > 7)
                       && abs(eg_value(pos.psq_score())) * 5 > (856 + pos.non_pawn_material() / 64) * (10 + pos.rule50_count()));
```
w/o the extra ? true : .

Reply:
The indentation looks rather strange?

Reply:
@mstembera, your indentation looks rather strange. I'd write it like this:
```
bool useClassical = !useNNUE ||
  ((pos.this_thread()->depth > 9 || pos.count<ALL_PIECES> () > 7) &&
    abs(eg_value(pos.psq_score())) *5 > (856 + pos.non_pawn_material() / 64) *(10 + pos.rule50_count()));
```

Reply:
Yes the indentation in master is strange too.  I have no actual preference for the indentation.

Reply:
@mstembera, don't get me wrong, though – your solution is better. We just have to properly indent the code. I don't know what code style does Stockfish follow, if any.

Reply:
@mstembera Yes, writing it that way is better.